Sources/SnapshotTesting/AssertSnapshot.swift

```diff
@@ -173,6 +173,7 @@ public func verifySnapshot<Value, Format>(
   )
   -> String? {
 
+    TestObserver.registerIfNeeded()
```

Comment:
Maybe the name of the type or its function could reflect why this observer registers itself and why it clears the `counterMap` variable.

Reply:
How about now? Does it sound better?